mintpy/prep_hyp3.py

```diff
@@ -0,0 +1,221 @@
+          dem.tif (a DEM with any name)
+
+  After running prep_hyp3.py:
+      For each interferogram:
+          S1AA_20161223T070700_20170116T070658_VVP024_INT80_G_ueF_74C2_unw_phase.tif
+          S1AA_20161223T070700_20170116T070658_VVP024_INT80_G_ueF_74C2_unw_phase.tif.rsc
+          S1AA_20161223T070700_20170116T070658_VVP024_INT80_G_ueF_74C2_corr.tif
+          S1AA_20161223T070700_20170116T070658_VVP024_INT80_G_ueF_74C2_corr.tif.rsc
+          S1AA_20161223T070700_20170116T070658_VVP024_INT80_G_ueF_74C2.txt
+      For the input DEM geometry file:
+          dem.tif
+          dem.tif.rsc
+
+  Notes:
+    HyP3 currently only supports generation of Sentinel-1 interferograms, so
+    some Sentinel-1 metadata is hard-coded. If HyP3 adds processing of interferograms
+    from other satellites, changes will be needed. 
+"""
+
+#########################################################################
+
+# CMD parsing
+def create_parser():
+    parser = argparse.ArgumentParser(description='Prepare attributes file for HyP3 InSAR product.\n'+
+                                     DESCRIPTION,
+                                     formatter_class=argparse.RawTextHelpFormatter,
+                                     epilog=EXAMPLE)
+
+    parser.add_argument('file', nargs='+', help='HyP3 file(s)')
+    return parser
+
+
+def cmd_line_parse(iargs=None):
+    parser = create_parser()
+    inps = parser.parse_args(args=iargs)
+    inps.file = ut.get_file_list(inps.file, abspath=True)
+    return inps
+
+
+#########################################################################
+
+# Extract geospatial and sensor metadata
+def general_metadata(fname):
+    '''Read/extract attribute data from HyP3 gdal geotiff metadata file and add to metadata dictionary
+    Inputs:
+        *unw_phase.tif or *corr.tif file name or *dem.tif file name, e.g.
+            S1AA_20161223T070700_20170116T070658_VVP024_INT80_G_ueF_74C2_unw_phase.tif
+            S1AA_20161223T070700_20170116T070658_VVP024_INT80_G_ueF_74C2_corr.tif
+            dem.tif
+    Output:
+        Metadata dictionary (meta)
+    '''
+
+    meta = {}
+
+    # Geospatial Data
+    ds = gdal.Open(fname, gdal.GA_ReadOnly)
+    meta['FILE_LENGTH'] = ds.RasterYSize
+    meta['LENGTH'] = ds.RasterYSize
+    meta['WIDTH'] = ds.RasterXSize
+    geoTrans = ds.GetGeoTransform()
+    del ds
+
+    x0 = geoTrans[0]
+    y0 = geoTrans[3]
+    x_step = geoTrans[1]
+    y_step = geoTrans[5]
+    x1 = x0 + x_step * meta['WIDTH']
+    y1 = y0 + y_step * meta['LENGTH']
+
+    meta['X_FIRST'] = f'{x0:.9f}'
+    meta['Y_FIRST'] = f'{y0:.9f}'
+    meta['X_STEP'] = f'{x_step:.9f}'
+    meta['Y_STEP'] = f'{y_step:.9f}'
+    meta['X_UNIT'] = 'METERS'
+    meta['Y_UNIT'] = 'METERS'
+
+    meta["LON_REF1"] = x0
+    meta["LON_REF2"] = x1
+    meta["LON_REF3"] = x0
+    meta["LON_REF4"] = x1
+
+    meta["LAT_REF1"] = y0
+    meta["LAT_REF2"] = y0
+    meta["LAT_REF3"] = y1
+    meta["LAT_REF4"] = y1
+
```

Comment:
`LAT/LON_REF1/2/3/4` are in a different order between ascending and descending datasets, as shown here: https://github.com/yunjunz/MintPy/blob/main/mintpy/prep_gmtsar.py#L103.

Could you please also fix this bug in `prep_aria.py` here?

Reply:
Just to be clear, would you like me to fix this bug for both prep_hyp3 and prep_aria in this PR?

Reply:
Also, should this change also be made for the geometry (DEM) dataset as well with the ascending/descending information from one of the interferograms? Out of curiosity, why is the order different for ascending and descending datasets?

Reply:
> Just to be clear, would you like me to fix this bug for both prep_hyp3 and prep_aria in this PR?

Yes please.

Reply:
> should this change also be made for the geometry (DEM) dataset as well with the ascending/descending information from one of the interferograms? 

Yes, it's should be in the metadata of all files.

From the strict [definition of REF_LAT/LON1/2/3/4](https://mintpy.readthedocs.io/en/latest/api/attributes/), they are in the order of first line near/far range, then last line near/far range, thus, for the same area, they are different between asc and desc.